simulator-starter/src/main/java/org/citrusframework/simulator/service/ScenarioExecutionService.java

```diff
@@ -55,18 +56,24 @@ public class ScenarioExecutionService implements DisposableBean, ApplicationList
     private final ApplicationContext applicationContext;
     private final Citrus citrus;
 
-    private final ThreadFactory threadFactory = new ThreadFactoryBuilder()
-            .setDaemon(true)
-            .setNameFormat("execution-svc-thread-%d")
-            .build();
-
-    private final ExecutorService executorService = Executors.newFixedThreadPool(10, threadFactory);
+    private final ExecutorService executorService;
 
     @Autowired
-    public ScenarioExecutionService(ActivityService activityService, ApplicationContext applicationContext, Citrus citrus) {
+    public ScenarioExecutionService(ActivityService activityService, ApplicationContext applicationContext,
+                                    Citrus citrus, SimulatorConfigurationProperties properties) {
         this.activityService = activityService;
         this.applicationContext = applicationContext;
         this.citrus = citrus;
+
+        ThreadFactory threadFactory = new ThreadFactoryBuilder()
+            .setDaemon(true)
+            .setNameFormat("execution-svc-thread-%d")
+            .build();
+
+        this.executorService = Executors.newFixedThreadPool(
+            properties.getExecutorThreads(),
+            threadFactory
```

Comment:
you could even inline the `ThreadFactory`.

Reply:
done.